Add building status to JobBuild

macklinu opened the pull request:
This commit adds support for a currently building Jenkins job
for the JobBuild widget.

Resolves #14

I figured making the change and getting a couple screenshots in front of you would help me show what I was thinking for this feature.
### Building

<img width="424" alt="2016-03-29 at 5 09 pm building" src="https://cloud.githubusercontent.com/assets/2344137/14124080/bc3b523e-f5d1-11e5-93ef-e04de8eaf991.png">
### Success

<img width="421" alt="2016-03-29 at 5 10 pm success" src="https://cloud.githubusercontent.com/assets/2344137/14124079/bc3b4cf8-f5d1-11e5-94db-08574386a75e.png">

Commit:
Add building status to JobBuild

This commit adds support for a currently building Jenkins job
for the JobBuild widget.

Resolves #14

plouc commented on src/components/JobBuild.jsx:
```diff
@@ -18,12 +18,27 @@ class JobBuild extends Component {
         );
     }
+
+    buildResult() {
```
I'm planing on moving a lot of the actual Mozaïk components to simple functions instead of classes, having no context could help, so perhaps this utility should be outside the component, for example:

``` javascript
const JENKINS_BUILD_STATUS_BUILDING = 'BUILDING';
const JENKINS_BUILD_STATUS_UNKNOWN  = 'UNKNOWN';
// perhaps other statuses could be added as constants

/**
 * Returns a build status from given build info.
 *
 * @param {object} build - the jenkins build object
 * @returns {string}
 */
const getBuildStatus = (build) => {
    if (build.result) {
        return build.result;
    }

    return build.building ? JENKINS_BUILD_STATUS_BUILDING : JENKINS_BUILD_STATUS_UNKNOWN;
};
```

macklinu replied:
That makes sense. I think breaking this out into its own component will also make it easier to reuse this functionality for other Jenkins widgets (assuming that the histogram and job status widgets encounter the same data from the Jenkins API).

Maybe it would even make sense down the road to extract models or decorators around Jenkins' job or build API objects that encapsulate this logic, allow for easy testing, and allow for reuse throughout the codebase. I don't have too much familiarity with React or JavaScript patterns in this sense, but something like:

``` es6
class BuildDecorator {

    const JENKINS_BUILD_STATUS_BUILDING = 'BUILDING';
    const JENKINS_BUILD_STATUS_UNKNOWN  = 'UNKNOWN';

    constructor(build) {
        this.build = build;
    }

    getBuildStatus() {
        if (this.build.result) {
            return this.build.result;
        }

        return this.build.building ? JENKINS_BUILD_STATUS_BUILDING : JENKINS_BUILD_STATUS_UNKNOWN;
    };
}
```

plouc replied:
@macklinu, I'll go for the simple function without context, it's easier to test and use IHMO.

macklinu replied:
Sounds good. I'm busy with client work at the moment but will make the change this week and ping you when a new commit has been pushed up. :+1:

plouc commented:
@macklinu, really nice!

Commit:
Extract getBuildStatus function

macklinu commented:
@plouc sorry for the delay on this one! Rebased against master to get the new ava test runner in my branch and extracted that `getBuildStatus` function. I didn't know where to put it, so I made a `src/components/util.js` file for now. If that's good, then 👍 , but if you have feedback on where those sorts of helpers will live, let me know and I'll make a quick update. Thanks!

plouc commented on styl/components/_list.styl:
```diff
@@ -0,0 +1,5 @@
+.list
```
@macklinu, I'll directly add the `&--building` modifier to Mozaïk, could be useful for other widgets, see https://github.com/plouc/mozaik/blob/master/src/styl/components/list.styl#L52

plouc replied:
So you can remove all specific css :-)

macklinu replied:
Removed in c7f0a61

plouc commented on test/components/JobBuild.test.js:
```diff
@@ -16,3 +17,16 @@ test('should display job build number and status', t => {
     t.regex(wrapper.text(), new RegExp(`#${build.number} ${build.result}`));
 });
+
+test('should display building job build number and status', t => {
```
:ok_hand:

Commit:
Remove JobBuilds specific CSS

macklinu commented:
@plouc any other updates I can make to this PR? Was reminded this was open by #18.

plouc commented:
Sorry for the delay @macklinu and thank you !

plouc commented:
@macklinu no, it's perfect, thank you and sorry for the delay.

macklinu commented:
Not a problem, thank you! :D

On Mon, May 23, 2016 at 11:18 AM, Raphaël Benitte notifications@github.com
wrote:

> @macklinu https://github.com/macklinu no, it's perfect, thank you and
> sorry for the delay.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/plouc/mozaik-ext-jenkins/pull/15#issuecomment-221008178

## 

Macklin Underdown
macklinu@gmail.com
